demos/adventure/adventure.go

```diff
@@ -0,0 +1,393 @@
+		spriteSheet,
+	)
+
+	hero.ControlComponent = ControlComponent{
+		SchemeHoriz: "horizontal",
+		SchemeVert:  "vertical",
+	}
+
+	hero.RenderComponent.SetZIndex(1)
+
+	// Add our hero to the appropriate systems
+	for _, system := range w.Systems() {
+		switch sys := system.(type) {
+		case *common.RenderSystem:
+			sys.Add(
+				&hero.BasicEntity,
+				&hero.RenderComponent,
+				&hero.SpaceComponent,
+			)
+
+		case *common.AnimationSystem:
+			sys.Add(
+				&hero.BasicEntity,
+				&hero.AnimationComponent,
+				&hero.RenderComponent,
+			)
+
+		case *ControlSystem:
+			sys.Add(
+				&hero.BasicEntity,
+				&hero.AnimationComponent,
+				&hero.ControlComponent,
+				&hero.SpaceComponent,
+			)
+		}
+	}
+
+	// Create render and space components for each of the tiles
+	tileComponents := make([]*Tile, 0)
+
+	for _, tileLayer := range levelData.TileLayers {
+		for _, tileElement := range tileLayer.Tiles {
+
+			if tileElement.Image != nil {
+				tile := &Tile{BasicEntity: ecs.NewBasic()}
+				tile.RenderComponent = common.RenderComponent{
+					Drawable: tileElement,
+					Scale:    engo.Point{1, 1},
+				}
+				tile.SpaceComponent = common.SpaceComponent{
+					Position: tileElement.Point,
+					Width:    0,
+					Height:   0,
+				}
+
+				if tileLayer.Name == "grass" {
+					tile.RenderComponent.SetZIndex(0)
+				}
+
+				if tileLayer.Name == "trees" {
+					tile.RenderComponent.SetZIndex(2)
+				}
+
+				tileComponents = append(tileComponents, tile)
+			}
+		}
+	}
+
+	for _, imageLayer := range levelData.ImageLayers {
+		for _, imageElement := range imageLayer.Images {
+
+			if imageElement.Image != nil {
+				tile := &Tile{BasicEntity: ecs.NewBasic()}
+				tile.RenderComponent = common.RenderComponent{
+					Drawable: imageElement,
+					Scale:    engo.Point{1, 1},
+				}
+				tile.SpaceComponent = common.SpaceComponent{
+					Position: imageElement.Point,
+					Width:    0,
+					Height:   0,
+				}
+
+				if imageLayer.Name == "clouds" {
+					tile.RenderComponent.SetZIndex(3)
+				}
+
+				tileComponents = append(tileComponents, tile)
+			}
+		}
+	}
+
+	// Add each of the tiles entities and its components to the render system
+	for _, system := range w.Systems() {
+		switch sys := system.(type) {
+		case *common.RenderSystem:
+			for _, v := range tileComponents {
+				sys.Add(&v.BasicEntity, &v.RenderComponent, &v.SpaceComponent)
+			}
+
+		}
+	}
+
+	// Access Object Layers
+	for _, objectLayer := range levelData.ObjectLayers {
+		log.Println("This object layer is called " + objectLayer.Name)
+		// Do something with every regular Object
+		for _, object := range objectLayer.Objects {
+			log.Println("This object is called " + object.Name)
+		}
+
+		// Do something with every polyline Object
+		for _, polylineObject := range objectLayer.PolyObjects {
+			log.Println("This object is called " + polylineObject.Name)
+		}
+	}
+
+	// Setup character and movement
+	engo.Input.RegisterAxis(
```

Comment:
Sins both wasd and arrow keys are used to rotate the character the wasd key pairs should be added to this axis and the one down below so that movement works as well.